PeerioTechnologies/peerio-desktop
# fix: on Windows use %LOCALAPPDATA% instead of default %APPDATA% for storing local files

The default path was storing files in the Roaming directory. We changed
it to Local. We also migrate from the old place to the new one by moving
the whole directory if the new directory doesn't exist.

#### Relevant info and issue/PR links 

https://peerio.tpondemand.com/entity/7785 
https://app.clubhouse.io/peerio/story/712/desktop-windows-user-data-folder-is-in-roaming

#### Testing instructions  

Testing is required on Windows only (Win7 too please).

A.

1. Before starting the app, in Windows Explorer open `AppData\Roaming` for the user and check that "Peerio 2" folder is there, and in another window `AppData\Local` and notice that "Peerio 2" folder is *not* there (this things can be typed right into the path bar).

2. Start the app.

3. Notice that "Peerio 2" folder is moved from `AppData\Roaming` to `AppData\Local` (if not, it's a bug).

4. Quit the app and start it again.

5. Check that "Peerio 2" folder is still in `AppData\Local` and there's no such folder in the `AppData\Roaming`.

Also, we should thing about failures that can happen. (If the folder fails to move for some reason, Peerio will continue using the old `Roaming` folder.)

B.

On a "freshly installed app", which doesn't have the corresponding data folder in `AppData\Roaming` or `AppData\Local`, make sure when the app starts, the data is saved to `AppData\Local` folder, not the roaming one.


----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
